vpn/tunnel.go

```diff
@@ -170,6 +288,60 @@ func sinkEntryToPb(e slog.SinkEntry) *Log {
 	return l
 }
 
+func convertWorkspaceUpdate(update tailnet.WorkspaceUpdate) *PeerUpdate {
+	out := &PeerUpdate{
+		UpsertedWorkspaces: make([]*Workspace, len(update.UpsertedWorkspaces)),
+		UpsertedAgents:     make([]*Agent, len(update.UpsertedAgents)),
+		DeletedWorkspaces:  make([]*Workspace, len(update.DeletedWorkspaces)),
+		DeletedAgents:      make([]*Agent, len(update.DeletedAgents)),
+	}
+	for i, ws := range update.UpsertedWorkspaces {
+		out.UpsertedWorkspaces[i] = &Workspace{
+			Id:     tailnet.UUIDToByteSlice(ws.ID),
+			Name:   ws.Name,
+			Status: Workspace_Status(ws.Status),
+		}
+	}
+	for i, agent := range update.UpsertedAgents {
+		fqdn := make([]string, 0, len(agent.Hosts))
+		for name := range agent.Hosts {
+			fqdn = append(fqdn, name.WithoutTrailingDot())
```

Comment:
We probably want these with the trailing dots.  I know you don't see domain names that often with the trailing dot, but what it means is that the domain name is fully qualified: that is, you shouldn't try to apply search suffixes.  In this case, we know that these names are fully qualified because Coder is also the DNS server that they refer to.